src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,194 @@
+import { DropdownStyle, OptionGroupStyle, OptionStyle } from './styles';
+import { SizeType } from '../../core/types/size';
+import IonIcon from '../icons';
+import React, { useState } from 'react';
+
+type SelectSizes = Extract<SizeType, 'sm' | 'md'>;
+
+export interface OptionProps {
+  disabled?: boolean;
+  label: string;
+  selected?: boolean;
+  value: string | number;
+}
+
+export interface OptionGroupProps {
+  disabled?: boolean;
+  options: OptionProps[];
+  label: string;
+}
+
+export interface DropdownProps {
+  disabled?: boolean;
+  multiple?: boolean;
+  options?: OptionProps[];
+  optionsGroup?: OptionGroupProps[];
+  selectedOptions?: OptionProps[];
+  size?: SelectSizes;
+  onSelectedOption: Function;
+}
+
+export interface OptionsConfig extends DropdownProps {
+  indexGroup?: number;
+}
+
+const isDisabled = (
+  disabledThisOption?: boolean,
+  disabledAllOptions?: boolean
+): boolean => {
+  const result = disabledAllOptions || disabledThisOption;
+  return result || false;
+};
+const isSelectingNotMultiple = (
+  isSelecting?: boolean,
+  multiple?: boolean
+): boolean => {
+  const result = !isSelecting && !multiple;
+  return result || false;
```

Comment:
invert here.

Reply:
I've tryna change name and values, but still, just works to select. When unseleting some option, it cracks, cuz, on not multiple mode, last option selected never will be unselect.